python/sidecar/chat.py

```diff
@@ -39,6 +59,7 @@ def get_relevant_documents(self):
         docs = self.ranker.get_top_n(query=self.input_text, limit=5)
         return docs
 
+    @retry(stop=stop_after_attempt(3), wait=wait_fixed(1))
```

Comment:
I wasn't quite sure how we'd want to handle retries, but this felt like a fine starting point. In reality, I imagine if we need to spend more than a few second retrying (e.g. the user is rate limited or the service is down), we probably do not want to be retrying at all. Our user is still trying to write a document within Refstudio, even if the AI functionality is not working.